apps/api/src/app/events/usecases/send-message/send-message.usecase.ts

```diff
@@ -60,6 +61,24 @@ export class SendMessage {
     const shouldRun = await this.filter(command);
     const preferred = await this.filterPreferredChannels(command.job);
 
+    if (!command.payload?.$on_boarding_trigger) {
+      this.analyticsService.track('Process Workflow Step - [Triggers]', command.userId, {
+        _template: command.job._templateId,
+        _organization: command.organizationId,
+        _environment: command.environmentId,
+        _subscriber: command.job?._subscriberId,
+        provider: command.job?.providerId,
+        delay: command.job?.delay,
+        jobType: command.job?.type,
+        digestType: command.job.digest?.type,
+        digestEventsCount: command.job.digest?.events?.length,
+        digestUnit: command.job.digest?.unit,
+        digestAmount: command.job.digest?.amount,
+        filterPassed: shouldRun,
+        preferencesPassed: preferred,
+      });
+    }
```

Comment:
I moved this event to be a little more above, like rn it will be sent event when the filters pass or not.
It's updated with additional information like environment, filters passed and preferences passed flags.